logstash-core/src/main/java/org/logstash/config/ir/SourceMetadata.java

```diff
@@ -40,6 +48,10 @@ public SourceMetadata() {
         this.sourceText = null;
     }
 
+    public int hashCode() {
+        return Objects.hash(this.sourceFile, this.sourceLine, this.sourceColumn, this.sourceText);
+    }
+
     public String toString() {
         return sourceFile + ":" + sourceLine + ":" + sourceColumn + ":```\n" + sourceText + "\n```";
     }
```

Comment:
We have decided to extract the `@JsonProperty` Does this PR need to be updated with that logic or we want to make it another PR for that.